lib/http/http_mock.dart

```diff
@@ -124,8 +125,17 @@ class HttpMock implements Http {
       if (response is Exception) {
         throw response;
       }
-
-      return Response(null, response);
+      http.Response httpResponse;
+      Map<String, dynamic> json;
+
+      if(response is Map<String, dynamic>) {
+        // The response is parsed json
+        json = response;
+      } else if (response is List<int>) {
+        // The response is a binary stream
+        httpResponse = http.Response.bytes(response, 200);
+      }
+      return Response(httpResponse, json);
```

Comment:
I think that some sort of error handling should be added (such that the method can never return Response(null, null), unless this actually should be allowed).
```
else
  throw SomeException;
```